agent/src/agent/cli/streamsets.py

```diff
@@ -0,0 +1,82 @@
+import click
+import requests
+
+from typing import List
+from agent import destination, pipeline
+from agent.pipeline.streamsets import StreamSets
+from agent.pipeline import streamsets, Pipeline
+from agent.modules import constants
+from agent.pipeline.streamsets.repository import StreamsetsNotExistsException
+from agent.modules.tools import infinite_retry
+
+
+def get_url_complete(ctx, args, incomplete):
+    return [s.url for s in streamsets.repository.get_all() if incomplete in s.url]
+
+
+@click.group(name='streamsets')
+def streamsets_group():
+    pass
+
+
+@click.command()
+def add():
+    _prompt_streamsets(StreamSets('', '', ''))
+    if destination.repository.exists():
+        pipeline.manager.create_monitoring_pipelines()
+
+
+@click.command()
+@click.argument('url', autocompletion=get_url_complete)
+def edit(url):
+    try:
+        s = streamsets.repository.get_by_url(url)
+    except StreamsetsNotExistsException:
+        raise click.UsageError(f'StreamSets with url {url} does not exist')
+    _prompt_streamsets(s)
+
+
+@click.command()
+@click.argument('url', autocompletion=get_url_complete)
+def delete(url):
+    try:
+        streamsets_ = streamsets.repository.get_by_url(url)
+        pipelines = pipeline.repository.get_by_streamsets(streamsets_)
+        if _has_pipelines(pipelines):
+            # -1 because we don't count Monitoring
+            if not click.confirm(f'Streamsets with url {streamsets_.url} contains {len(pipelines) - 1} pipelines, all these pipelines will be DELETED, continue?'):
+                return
+            for pipeline_ in pipelines:
+                pipeline.repository.delete(pipeline_)
```

Comment:
Don't delete pipelines, move them to a different streamsts if available. I no streamsets available, don't delete anything